Use fantasticon to generate web fonts

mdo opened the pull request:
Replaces #287 with a new dependency, fantasticon. Fixes #86.

XhmikosR commented on docs/assets/scss/docs.scss:
```diff
@@ -140,3 +140,29 @@
 }
 // stylelint-enable declaration-no-important
+
+// stylelint-disable
```
Better be more explicit here?

mdo replied:
Done, but it's a mouthful haha.

XhmikosR replied:
Indeed, better revert this one, sorry!

XhmikosR commented on .fantasticonrc.js:
```diff
@@ -0,0 +1,30 @@
+    woff2: './font/fonts/bootstrap-icons.woff2',
+    eot: './font/fonts/bootstrap-icons.eot'
+  }
+};
```
EOF newline

XhmikosR commented on src/html.hbs:
```diff
@@ -0,0 +1,53 @@
+  </div>
+
+</body>
+</html>
```
EOF newline

XhmikosR commented on package.json:
```diff
@@ -61,7 +62,8 @@
     "icons/*.svg",
-    "bootstrap-icons.svg"
+    "bootstrap-icons.svg",
+    "fonts"
```
Better be more explicit here like this:

```suggestion
    "fonts/*.{eot,ttf,woff,woff2}"
```

mdo replied:
I want the whole fonts directory for now. Too weird?

XhmikosR replied:
Yup. Because unneeded files might be included in the npm package, even files not tracked by git.

XhmikosR replied:
Actually, now that I see the folder contents, ignore my comment. It's just too many filetypes to specify.

XhmikosR reviewed (changes requested):
Also,

* the hbs files seem unused
* package.json is missing the new devDependency
* the script should be added to another script (or we run it separately) so that's tested on CI

mdo commented:
> * the hbs files seem unused

`.hbs` files are indeed used—they help generate the CSS and HTML in the `font` directory. They're specified in `.fantasticonrc.js`.

mdo commented:
> * package.json is missing the new devDependency
> * the script should be added to another script (or we run it separately) so that's tested on CI

Done and done.

- Hooked into the `icons` script
- Added devDependency
- Renamed the `font` script to `icons-font` for consistency
- Added the `font` dir to the zip script

XhmikosR commented:
Just a question: do you really want to add eot and ttf fonts? woff and woff2 should be more than enough. Heck, you could even just generate woff2 if you don't care about IE :P

* https://caniuse.com/?search=woff
* https://caniuse.com/?search=woff2

XhmikosR commented on src/html.hbs:
```diff
@@ -0,0 +1,53 @@
+<!doctype html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
```
Maybe lowercase this?

XhmikosR commented on src/html.hbs:
```diff
@@ -0,0 +1,53 @@
+    .icons {
+      display: grid;
+      max-width: 100%;
+      grid-template-columns: repeat(auto-fit, minmax(100px, 1fr) );
```
Shouldn't this be prefixed?

mdo replied:
I don't think so? https://deploy-preview-31813--twbs-bootstrap.netlify.app/docs/5.0/layout/css-grid/ doesn't show any prefixing...

XhmikosR replied:
Yeah, it needs to be manually made since that's not handled by any tools: it's hardcoded.

XhmikosR replied:
NVM, sorry, I just re-read what you wrote. Indeed, it's fine as is :)

mdo commented:
Dropping EOT and TTF for now since we can always add back later, but removing is always more difficult.

Good to go after this I think!

XhmikosR commented:
Minus my last comment, I think this looks good :)

XhmikosR commented:
I wonder though, is fonts/index.html supposed to be served? It's not using relative paths to the CSS AFAICT.

mdo commented:
> I wonder though, is fonts/index.html supposed to be served? It's not using relative paths to the CSS AFAICT.

Yeah, I want that served. It's used locally for previewing, and eventually I'll have it up to date and part of the full nav.

XhmikosR commented:
Actually, I just saw you hooked the font directory as font so it's already available indeed: https://deploy-preview-515--bootstrap-icons.netlify.app/font/ and included in sitemap.

TechQuery commented:
@mdo @XhmikosR There is no **CSS & Web Font** files in the [ZIP bundle][1] & [NPM package][2]...

[1]: https://github.com/twbs/icons/releases/download/v1.2.0/bootstrap-icons-1.2.0.zip
[2]: https://www.jsdelivr.com/package/npm/bootstrap-icons

XhmikosR commented:
The zip file does have the font for me. For the npm package see #538 .
